packages/react-native-quick-crypto/cpp/cipher/HybridCipher.cpp

```diff
@@ -80,17 +80,20 @@ bool HybridCipher::initAuthenticated(
   }
 
   const int mode = getMode();
-  if (mode == EVP_CIPH_GCM_MODE || mode == EVP_CIPH_OCB_MODE) {
-    // Set default tag length for OCB mode if not specified
-    if (mode == EVP_CIPH_OCB_MODE && auth_tag_len == kNoAuthTagLength) {
-      auth_tag_len = 16;  // Default to 16 bytes for OCB
+  if (mode == EVP_CIPH_GCM_MODE || mode == EVP_CIPH_OCB_MODE || mode == EVP_CIPH_SIV_MODE) {
+    // Set default tag length for OCB and SIV modes if not specified
+    if ((mode == EVP_CIPH_OCB_MODE || mode == EVP_CIPH_SIV_MODE) && auth_tag_len == kNoAuthTagLength) {
+      auth_tag_len = 16;  // Default to 16 bytes for OCB and SIV
     }
 
-    // Both GCM and OCB modes use similar tag length validation
+    // GCM, OCB and SIV modes use similar tag length validation
     if (!isValidAEADTagLength(auth_tag_len, mode)) {
-      throw std::runtime_error("Invalid authentication tag length (" +
-        std::string(mode == EVP_CIPH_GCM_MODE ? "GCM" : "OCB") +
-        ")");
+      std::string mode_str;
+      if (mode == EVP_CIPH_GCM_MODE) mode_str = "GCM";
+      else if (mode == EVP_CIPH_OCB_MODE) mode_str = "OCB";
+      else mode_str = "SIV";
```

Comment:
Could this be a switch statement instead?

https://www.w3schools.com/cpp/cpp_switch.asp